jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererRecommendationManager.kt

```diff
@@ -141,7 +141,7 @@ class CodeWhispererRecommendationManager {
             }
 
             val reformatted = reformat(requestContext, truncated)
-            val isDiscardedByRightContextTruncationDedupe = truncated.content().isEmpty() || !seen.add(reformatted.content())
+            val isDiscardedByRightContextTruncationDedupe = truncated.content().isEmpty() || !seen.add(reformatted.content()) || truncatedContent == userInput
```

Comment:
So what's been done here is to check if the userinput is equal to the suggestion, after truncation. This actually means we should also do this if there's no truncation, since in the above we also just do `!it.content().startsWith(userInput)` but didn't include the strict equal case.

Reply:
this is on line 122.

Reply:
yes make sense, updated but I think we still need to keep the condition `!it.content().startsWith(userInput)` right? @andrewyuq